packages/myst-transforms/src/enumerate.ts

```diff
@@ -435,7 +449,7 @@ export class MultiPageReferenceState implements IReferenceState {
 
 export const enumerateTargetsTransform = (tree: Root, opts: StateOptions) => {
   opts.state.initializeNumberedHeadingDepths(tree);
-  const nodes = selectAll('container,math,heading,[identifier]', tree) as (
+  const nodes = selectAll('container,math,heading,proof,[identifier]', tree) as (
```

Comment:
We can do a bit better in future about making this extensible, but this gets the job done for now!